src/app/core/services/local-storage.service.ts

```diff
@@ -0,0 +1,12 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({ providedIn: 'root' })
+export class LocalStorageService {
+  static setSavedState(state: any, localStorageKey: string) {
+    localStorage.setItem(localStorageKey, JSON.stringify(state));
+  }
+
+  static getSavedState(localStorageKey: string): any {
```

Comment:
is it really necessary? type-unsafe, imho

Reply:
Yep, urr, I've just fixed it!